specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/1.2-preview/iotcentral.json

```
@@ -0,0 +1,5638 @@
+    },
+    "DeviceId": {
+      "in": "path",
+      "name": "deviceId",
+      "description": "Unique ID of the device.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": true
+    },
+    "DeviceTemplateId": {
+      "in": "path",
+      "name": "deviceTemplateId",
+      "description": "Unique [Digital Twin Model Identifier](https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/dtdlv2.md#digital-twin-model-identifier) of the device template.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": true
+    },
+    "ExportId": {
+      "in": "path",
+      "name": "exportId",
+      "description": "Unique ID for the export.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": true
+    },
+    "Filter": {
+      "in": "query",
+      "name": "$filter",
+      "description": "An expression on the resource type that selects the resources to be returned.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": false
+    },
+    "JobId": {
+      "in": "path",
+      "name": "jobId",
+      "description": "Unique ID of the job.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": true
+    },
+    "ModuleName": {
+      "in": "path",
+      "name": "moduleName",
+      "description": "Name of the device module.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": true
+    },
+    "OrderBy": {
+      "in": "query",
+      "name": "$orderby",
+      "description": "An expression that specify the order of the returned resources.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": false
+    },
+    "RelationshipId": {
+      "in": "path",
+      "name": "relationshipId",
+      "description": "Unique ID of a relationship between devices.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": true
+    },
+    "Subdomain": {
+      "in": "path",
+      "name": "subdomain",
+      "description": "The application subdomain.",
+      "x-ms-parameter-location": "client",
+      "type": "string",
+      "required": true
+    },
+    "TelemetryName": {
+      "in": "path",
+      "name": "telemetryName",
+      "description": "Name of this device telemetry.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": true
+    },
+    "Top": {
+      "in": "query",
+      "name": "$top",
+      "description": "The maximum number of resources to return from the collection.",
+      "x-ms-parameter-location": "method",
+      "type": "integer",
+      "required": false
+    },
+    "Version": {
+      "in": "query",
+      "name": "api-version",
+      "description": "The version of the API being called.",
+      "x-ms-parameter-location": "client",
+      "type": "string",
+      "required": true
+    }
+  },
```

You should also define a "security requirement" that applies the "azure_auth" security scheme to your operations. This is easy to do at the top level:
```suggestion
  },
  "security": [
    {
      "azure_auth": [ "user_impersonation" ]
    }
  ]
```